### DISCUSSION TITLE
ZeroNet client for Gemini protocol?

### DISCUSSION THREAD

**Discussion by d47081** [zeronet-conservancy/zeronet-conservancy#262] [General]
Gopher-like protocol, for small web. 

Thoughts it would be cool composition with ZeroNet / DHT hosting, but current client depends of HTML and JS.

What do you think about?

Details here:
https://geminiprotocol.net

**Comment by caryoscelus** [comment]
i think narrow focus is both its biggest strength *and* weakness at the same time. personally i prefer universal platform that allows specialized solutions on top of it. that said, it'd be cool to see two p2p projects grow together

on technical side, i'm not sure how possible integration could work, but if something needed on 0net side i'd be glad to help

**Reply by d47081** [reply]
These days, I understand that DHT/Bittorrent does not follow those privacy conceptions of Gemini protocol. It's pretty different networking models, that could be used together, but maybe not really good idea if we are talking about Gemini goals. So closing ZN+Gemini idea for myself.

Anyway, HTML/JS/Cookies running connections on background not something that I really want to use.

What about Riza project, is any news there?

**Reply by caryoscelus** [reply]
i'm currently concentrated (when working on these projects anyway) on making 0net usable, spam-resistant and more pleasant to develop for. the latter will in turn make it easier to later make js and browser optional. `riza` is currently more of a direction of 0net development. i'm hoping iterative design would bring us there, but it's something for the nearest future

**Comment by TokyoghoulEs** [comment]
Python version Which version of Python is recommended to use to run the project?

I have tried Python 3.8.x, 3.11.x, 3.12.x and I see no difference.

Thank you.

**Reply by slrslr** [reply]
I would use latest available unless other apps having issues with it.

**Reply by TokyoghoulEs** [reply]
> I would use latest available unless other apps having issues with it.

Ok

**Reply by caryoscelus** [reply]
you can use any, but i would also use the latest you can. we're saving 3.8 (?) compatibility basically for those old systems where you can't install newer python, but it's not guaranteed to last. development convenience should be more of a priority atm

PS: please create a separate discussion thread next time
